diff -urN a/gcc/config/i386/mingw32.h b/gcc/config/i386/mingw32.h
--- a/gcc/config/i386/mingw32.h	2017-01-17 09:23:40.000000000 +0200
+++ b/gcc/config/i386/mingw32.h	2020-03-26 10:28:15.000000000 +0200
@@ -172,7 +172,7 @@
 /* For native mingw-version we need to take care that NATIVE_SYSTEM_HEADER_DIR
    macro contains POSIX-style path.  See bug 52947.  */
 #undef NATIVE_SYSTEM_HEADER_DIR
-#define NATIVE_SYSTEM_HEADER_DIR "/mingw/include"
+#define NATIVE_SYSTEM_HEADER_DIR "/include"
 
 /* Output STRING, a string representing a filename, to FILE.
    We canonicalize it to be in Unix format (backslashes are replaced
diff -urN a/gcc/config.gcc b/gcc/config.gcc
--- a/gcc/config.gcc	2018-01-08 15:39:11.000000000 +0200
+++ b/gcc/config.gcc	2020-03-26 10:27:43.000000000 +0200
@@ -1771,7 +1771,7 @@
 			tmake_file="${tmake_file} i386/t-mingw-w32"
 			;;
 	esac
-        native_system_header_dir=/mingw/include
+        native_system_header_dir=/include
 	target_gtfiles="\$(srcdir)/config/i386/winnt.c"
 	extra_options="${extra_options} i386/cygming.opt i386/mingw.opt"
 	case ${target} in
diff -urN a/libcpp/lex.c b/libcpp/lex.c
--- a/libcpp/lex.c	2018-01-05 15:58:54.000000000 +0200
+++ b/libcpp/lex.c	2020-03-26 10:29:43.000000000 +0200
@@ -568,7 +568,7 @@
     {
       vc m_nl, m_cr, m_bs, m_qm;
 
-      data = *((const vc *)s);
+      data = __builtin_vec_vsx_ld (0, s);
       s += 16;
 
       m_nl = (vc) __builtin_vec_cmpeq(data, repl_nl);
